src/Jupyter/Visualization/StateDisplayOperations.cs

```diff
@@ -97,13 +98,35 @@ public override bool Dump(IQArray<Qubit>? qubits = null)
             // At this point, _data should be filled with the full state
             // vector, so let's display it, counting on the right display
             // encoder to be there to pack it into a table.
-            Channel.Display(new DisplayableState
+
+            var count = 0; //TODO: figure out how to increment
+            var state = new DisplayableState
             {
                 // We cast here as we don't support a large enough number
                 // of qubits to saturate an int.
                 QubitIds = qubits?.Select(q => q.Id) ?? Simulator.QubitIds.Select(q => (int)q),
                 NQubits = (int)_count,
-                Amplitudes = _data
+                Amplitudes = _data,
+                DivId = $"dump-machine-div-{count}" 
+            };
+            Channel.Display(state);
+
+            Channel.SendIoPubMessage(new Message
+            {
+                Header = new MessageHeader()
+                {
+                    MessageType = "iqsharp_state_dump"
+                },
+                Content = new MeasurementHistogramContent() //TODO: throwing error
+                {
+                    DivId = state.DivId,
+                    QubitIds = state.QubitIds,
+                    NQubits = state.NQubits,
+                    Amplitudes = state.SignificantAmplitudes(
```

Comment:
As discussed elsewhere, it would be good to make sure that _all_ amplitudes are passed to the client here.

Reply:
That makes sense. Sorry, I think I initially made the suggestion to use `SignificantAmplitudes` here, but I agree it would be nice for the client to get everything and be able to decide which amplitudes to display.